modules/Emails/include/ListView/ListViewDataEmailsSearchOnCrm.php

```diff
@@ -0,0 +1,316 @@
+     * @param InboundEmail $inboundEmail
+     * @param array $params
+     * @param Email $seed
+     * @param bool $singleSelect
+     * @param ??? $id
+     * @param int $limit
+     * @param User $currentUser
+     * @param string $idField
+     * @param int $offset
+     * @return array
+     */
+    public function search($filterFields, $request, $where, InboundEmail $inboundEmail, $params, Email $seed, $singleSelect, $id, $limit, User $currentUser, $idField, $offset) {
+        // Fix fields in filter fields
+
+        $filterFields = $this->lvde->fixFieldsInFilter($filterFields, $request, $where);
+
+
+        // Filter imported emails based on the UID of the results from the IMap server
+
+        if(!empty($where)) {
+            $where .= ' AND ';
+        }
+        $crmWhere = $where . 'mailbox_id LIKE ' .'"' . $inboundEmail->id . '"';
+
+
+        // Populates CRM fields
+
+        $crmQueryArray = $this->lvde->getCrmQueryArray($crmWhere, $filterFields, $params, $seed, $singleSelect);
+
+
+        // get crm emails query
+
+        $crmEmailsQuery = $this->lvde->getCrmEmailsQuery($crmQueryArray, $params);
+
+
+
+
+        $this->lvde->setVariableName($seed->object_name, $crmWhere, $this->lvde->listviewName, $id);
+
+        SugarVCR::erase($seed->module_dir);
+        $this->lvde->seed =& $seed;
+        $totalCounted = empty($GLOBALS['sugar_config']['disable_count_query']);
+        $_SESSION['MAILMERGE_MODULE_FROM_LISTVIEW'] = $seed->module_dir;
+        if(empty($request['action']) || $request['action'] != 'Popup'){
+            $_SESSION['MAILMERGE_MODULE'] = $seed->module_dir;
+        }
+
+        $this->lvde->setVariableName($seed->object_name, $where, $this->lvde->listviewName, $id);
+
+        $this->lvde->seed->id = '[SELECT_ID_LIST]';
+
+        // if $params tell us to override all ordering
+        if(!empty($params['overrideOrder']) && !empty($params['orderBy'])) {
+            $order = $this->lvde->getOrderBy(strtolower($params['orderBy']), (empty($params['sortOrder']) ? '' : $params['sortOrder'])); // retreive from $_REQUEST
+        }
+        else {
+            $order = $this->lvde->getOrderBy(); // retreive from $_REQUEST
+        }
+
+        // still empty? try to use settings passed in $param
+        if(empty($order['orderBy']) && !empty($params['orderBy'])) {
+            $order['orderBy'] = $params['orderBy'];
+            $order['sortOrder'] =  (empty($params['sortOrder']) ? '' : $params['sortOrder']);
+        }
+
+        if (empty($params['skipOrderSave'])) { // don't save preferences if told so
+            $currentUser->setPreference('listviewOrder', $order, 0, $this->lvde->var_name); // save preference
+        }
+
+        $data = array();
+
+        $rows = array();
+        $count = 0;
+        $idIndex = array();
+        $id_list = '';
+
+        $crmEmails = $this->lvde->db->query($crmEmailsQuery);
+
+        while(($row = $this->lvde->db->fetchByAssoc($crmEmails)) != null)
```

Comment:
(class ListViewDataEmailsSearchOnIMap and class ListViewDataEmailsSearchOnCrm not completely refactored yet)